MovieTime/MovieTime/Core/Extensions/Date/DateHelper.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  DateHelper.swift
+//  MovieTime
+//
+//  Created by Enrique Melgarejo on 10/01/18.
+//  Copyright © 2018 Choynowski. All rights reserved.
+//
+
+import Foundation
+
+class DateHelper {
+
+    static let shared = DateHelper()
```

Comment:
how to avoid using the singleton pattern for this entity? And what are the reasons to avoid this pattern?

Reply:
I know that I am violating the **S** (*Single responsibility principle*) of **SOLID** once they control their own creation and life cycle. In addition to that, they carry state around for the lifetime of the application, but I decided to use this pattern because of performance issues, since the *Calendar* and *DateFormatters* will be created only once and I want to guarantee that one and only one instance of a given class exists during run time.

Reply:
one of the biggest reasons to avoid a singleton is that they're hard to unit test. The fact that it breaks the single responsibility principle is just a purist look to the problem, at least I don't see an architectural problem only because a class can create itself

You could avoid it here by creating static properties inside an extension. It's a pattern that Apple uses a lot in their APIs. For example:
```
extension Calendar {
  static let defaultCalendar: Calendar = {
    var cal = Calendar.current
    cal.locale = Locale.current
    cal.timeZone = TimeZone.current
    return cal
  }()
}

let myCal = Calendar.defaultCalendar
```